src/Language/PureScript/CoreFn/ToJSON.hs

```diff
@@ -64,8 +64,8 @@ moduleToJSON v m = object [ T.pack "imports"   .= map (annToJSON *** moduleNameT
                           ]
 
 bindToJSON :: Bind Ann -> Value
-bindToJSON (NonRec ann n e) = object [ runIdent n .= ( annToJSON ann, exprToJSON e ) ]
-bindToJSON (Rec bs) = object $ map (\((ann, n), e) -> runIdent n .= ( annToJSON ann, exprToJSON e ) ) bs
+bindToJSON (NonRec ann n e) = toJSON [(runIdent n, annToJSON ann, exprToJSON e)]
+bindToJSON (Rec bs) = toJSON $ map (\((ann, n), e) -> (runIdent n, annToJSON ann, exprToJSON e)) bs
```

Comment:
I just couldn't compile the `Aff` module.  It contains a group of recursive binds with `applicativeAff`, `nonCanceler` and `alwaysCanceler`.  When reading it using a hash map the order was changed to `nonCanceler`, `alwaysCanceler` followed by `applicativeAff`.  ILooking at the compiled code one can see that here order is important:
```
var applicativeAff = new Control_Applicative.Applicative(function () {
    return applyAff;
}, function (v) {
    return $foreign._pure(nonCanceler, v);
});
var nonCanceler = Data_Function["const"](Control_Applicative.pure(applicativeAff)(false));
var alwaysCanceler = Data_Function["const"](Control_Applicative.pure(applicativeAff)(true));
```

When `applicativeAff` is at the end, executing this in `node` was failing with an error indicating that `applicativeAff` passed to `Control_Applicative.pure(applicativeAff)` in `nonCanceler` is `undefined`.